Adding the homepage Gallery

flukeout opened the pull request:
In-progress work for adding a project gallery to the homepage.
It's modeled after this protototype: https://flukeout.github.io/thimble-gallery/

Activity data lives in a separate repo, here:
* https://github.com/mozilla/thimble-homepage-gallery/blob/gh-pages/activities.json

**To-do**
* Responsive styling
* Handle cases where lots of tags in search box break the layout
* Rip out code related to old Starter Projects implementation
* Add loading indicator when external activity data is being fetched

@gideonthomas could you review the progress thus far?

Should appear like this, on the homepage...
![image](https://cloud.githubusercontent.com/assets/25212/23442613/72e0655a-fdde-11e6-9154-13dd006f7b0a.png)

flukeout commented:
Also @gideonthomas can you help me resolve the travis errors? Looks like it can't find some files. Do I have to run a build step of some sort in thimble?

gideonthomas commented on public/homepage/scripts/main.js:
```diff
@@ -17,6 +17,7 @@ require.config({
     "uuid": "/node_modules/node-uuid/uuid",
     "cookies": "/node_modules/cookies-js/dist/cookies",
     "analytics": "/node_modules/webmaker-analytics/analytics",
+    "gallery": "/{{ locale }}/homepage/scripts/gallery",
```
If you're creating it as a module here, you'll also need to add it to the Gruntfile here: https://github.com/mozilla/thimble.mozilla.org/blob/master/Gruntfile.js#L37-L50
`gallery: "homepage/scripts/gallery"`

flukeout commented:
Now it looks like travis is failing on a template literal...

``>> Unexpected character '`' (line: 1009, col: 21, pos: 116354)``

...is that right? Should I change this?

gideonthomas commented:
@flukeout yeah we don't support ES6 code yet. I would also suggest moving that to the top as a variable maybe.

flukeout commented:
Okay! Moved the template markup to the gallery.html page instead. So long, template literals.

gideonthomas commented:
@TheoChevalier would you be willing to look at the l10n part of this PR please? I'm a bit unsure about some of the strings (like "By", "Lesson Plan", etc.  and the way they're ordered)

TheoChevalier commented on views/gallery.html:
```diff
@@ -0,0 +1,48 @@
+          <span class='search-tags'></span>
+          <input class='search' placeholder='{{ gettext("gallerySearchPlaceholder") }}' type='text' />
+          <div class='icon search-icon'></div>
+          <div title='Clear the search term' class='clear icon clear-icon'></div>
```
Wrap `'Clear the search term'` for l10n

TheoChevalier commented on locales/en-US/messages.properties:
```diff
@@ -166,6 +166,17 @@ homeworkExcuseGeneratorRemixTitle=Remix the {{ homeworkExcuseGenerator }} projec
+lessonPlan=Lesson Plan
+projectBy=By
+remixGalleryTitle=Remix a project to get started...
+projectSingular=project
```
Let’s use something like `projectSingular={{ n }} project`, so that the number can be moved around.
Ideally, we should use proper plural form, since singular/plural won’t work for most languages:
`project[one]={{ n }} project`
`project[two]={{ n }} projects`
`project[few]={{ n }} projects`
`project[many]={{ n }} projects`

But I’m not sure what plural form support you have with this gettext implem (is this node-gettext?)

TheoChevalier commented on locales/en-US/messages.properties:
```diff
@@ -166,6 +166,17 @@ homeworkExcuseGeneratorRemixTitle=Remix the {{ homeworkExcuseGenerator }} projec
+projectBy=By
+remixGalleryTitle=Remix a project to get started...
+projectSingular=project
+projectPlural=projects
```
Same here, let’s use something like `projectPlural={{ n }} projects`

TheoChevalier commented on locales/en-US/messages.properties:
```diff
@@ -166,6 +166,17 @@ homeworkExcuseGeneratorRemixTitle=Remix the {{ homeworkExcuseGenerator }} projec
+remixGalleryTitle=Remix a project to get started...
+projectSingular=project
+projectPlural=projects
+found=found
```
If it’s concatenated with a number, we will need proper plural here, too. At least singular/plural depending on the support you have

flukeout commented:
Thanks for all of the great suggestions @TheoChevalier - I've removed the "Tags" content from the CSS entirely and reworked one of the headings to not rely on numbers so localization is easier.

TheoChevalier commented:
Thanks @flukeout, looks great!

flukeout commented:
Alright @gideonthomas want to look this over one last time?
